include/deal.II/base/polynomial.h

```diff
@@ -104,6 +104,20 @@ namespace Polynomials
                 std::vector<number> &values) const;
 
     /**
+     * Return the values and the derivatives of the Polynomial at point
+     * <tt>x</tt>.  <tt>values[i], i=0,...,values_size-1</tt> includes the
+     * <tt>i</tt>th derivative. The number of derivatives to be computed is
+     * determined by @p values_size and @p values has to provide sufficient
+     * space for @p values_size values.
+     *
+     * This function uses the Horner scheme for numerical stability of the
+     * evaluation.
+     */
+    void value (const number         x,
+                const unsigned int values_size,
+                number *values) const;
```

Comment:
I think it would make sense to call the variable indicating the length `n_derivatives` and state that the user has to make sure that `n_derivatives+1` entries are valid in the `values` array. While it's a nice concept to let the length of a `std::vector` determine the number of derivatives in the other call, one has to read twice in that variant, so here it definitely makes no sense to not use the name `n_derivatives`.